fix(docs): solve type checking errors in Markdown code samples

GJZwiers opened the pull request:
- Fixes issues found by running `deno test --doc --unstable` at the root of the manual
- Updates Deno.Signal examples in `examples/os_signals.md` to reflect changes made in 1.14.0

Notes:

- I changed the type of several code blocks in `runtime/workers.md` to `js` because `ts` throws the following errors when running `deno test --doc`:

```
TS2345 [ERROR]: Argument of type '{ type: "module"; deno: { namespace: boolean; permissions: { env: boolean; hrtime: boolean; net: string; ffi: boolean; read: boolean; run: boolean; write: boolean; }; }; }' is not assignable to 
parameter of type 'WorkerOptions'.                                                                                e: boolean; net: 
  Object literal may only specify known properties, and 'deno' does not exist in type 'WorkerOptions'.             'WorkerOptions'.
    deno: {
    ^
    at file:///path/to/my/dir/manual/runtime/workers.md$197-215.ts:4:5
```

There appear to be two interfaces called `WorkerOptions` in two different `.d.ts` files and it's using the wrong one somehow

- Examples with local paths such as `./foo.js` don't work with `deno test --doc` so I added ignores for now
- During local testing I replaced the `$STD_VERSION` vars with 0.107.0 to make it work. If they can also be handled in the CI pipeline then I believe `deno test --doc --unstable` can be added and run for every workflow

bartlomieju commented:
@GJZwiers thank you, I already started similar PR in https://github.com/denoland/manual/pull/32. Let me try to point you on a few things:

> I changed the type of several code blocks in runtime/workers.md to js because ts throws the following errors when running deno test --doc:

This appears to be caused by not using `--unstable` flag - `deno` object in `WorkerOptions` is an extension provided by Deno and is considered unstable. So you might want to try running with `--unstable` flag and reverting back to using `ts`.

> Examples with local paths such as ./foo.js don't work with deno test --doc so I added ignores for now

That's fine 👍 

> During local testing I replaced the $STD_VERSION vars with 0.107.0 to make it work. If they can also be handled in the CI pipeline then I believe deno test --doc --unstable can be added and run for every workflow

This can be solved by using an import map for testing; please refer to #32 to see how it's done.

GJZwiers commented:
I changed back the code blocks to `ts` and running that with `--unstable` works. Also fixed some more typechecking issues. Something that was also mentioned in the other PR is that the syntax highlighting goes away when using `ts, ignore` blocks, so I don't know if that's desirable to push through. 

Also encountered a weird error where `deno test --doc --unstable` panics, which would go away by changing LF to CRLF line endings in VS Code. I will try to see if I can reproduce that and if so make a separate issue. Otherwise I managed to get a clean run of `deno test --doc --unstable` locally with the current changes

GJZwiers commented:
In any case, feel free to combine the changes here with #32 if that's convenient.
